apps/test/unit/templates/codeDocs/ProgrammingClassOverviewTest.jsx

```diff
@@ -0,0 +1,150 @@
+      wrapper
+        .find('EnhancedSafeMarkdown')
+        .at(2)
+        .props().markdown
+    ).to.equal(defaultProgrammingClass.tips);
+    expect(
+      wrapper
+        .find('EnhancedSafeMarkdown')
+        .at(3)
+        .props().markdown
+    ).to.contain(defaultProgrammingClass.externalDocumentation);
+
+    expect(wrapper.text()).to.contain(defaultProgrammingClass.category);
+  });
+
+  it('hides the examples header if no syntax is provided', () => {
+    delete defaultProgrammingClass.examples;
+    const wrapper = shallow(
+      <ProgrammingClassOverview programmingClass={defaultProgrammingClass} />
+    );
+    expect(wrapper.find('h2').length).to.be.greaterThan(0);
+    expect(wrapper.find('h2').map(h => h.text())).to.not.include('Examples');
+  });
+
+  it('hides the syntax header if no syntax is provided', () => {
+    delete defaultProgrammingClass.syntax;
+    const wrapper = shallow(
+      <ProgrammingClassOverview programmingClass={defaultProgrammingClass} />
+    );
+    expect(wrapper.find('h2').length).to.be.greaterThan(0);
+    expect(wrapper.find('h2').map(h => h.text())).to.not.include('Syntax');
+  });
+
+  it('hides the fields headers if no fields are provided', () => {
+    defaultProgrammingClass.fields = [];
+    const wrapper = shallow(
+      <ProgrammingClassOverview programmingClass={defaultProgrammingClass} />
+    );
+    expect(wrapper.find('h2').length).to.be.greaterThan(0);
+    expect(wrapper.find('h2').map(h => h.text())).to.not.include('Fields');
+  });
+
+  it('hides the tips header if no tips is provided', () => {
+    delete defaultProgrammingClass.tips;
+    const wrapper = shallow(
+      <ProgrammingClassOverview programmingClass={defaultProgrammingClass} />
+    );
+    expect(wrapper.find('h2').length).to.be.greaterThan(0);
+    expect(wrapper.find('h2').map(h => h.text())).to.not.include('Tips');
+  });
+
+  it('hides the additional information header if no external documentation is provided', () => {
+    delete defaultProgrammingClass.externalDocumentation;
+    const wrapper = shallow(
+      <ProgrammingClassOverview programmingClass={defaultProgrammingClass} />
+    );
+    expect(wrapper.find('h2').length).to.be.greaterThan(0);
+    expect(wrapper.find('h2').map(h => h.text())).to.not.include(
+      'Additional Information'
+    );
+  });
+
+  it('uses color if color is provided', () => {
+    const wrapper = shallow(
+      <ProgrammingClassOverview
+        programmingClass={{
+          ...defaultProgrammingClass,
+          color: '#fff176'
+        }}
+      />
+    );
+    expect(wrapper.find('span').props().style.backgroundColor).to.equal(
+      '#fff176'
+    );
+  });
+
+  it('handles null color', () => {
+    const wrapper = shallow(
+      <ProgrammingClassOverview
+        programmingClass={{
+          ...defaultProgrammingClass,
+          color: null
+        }}
+      />
+    );
+    expect(wrapper.find('span').props().style.backgroundColor).to.be.null;
```

Comment:
Nit: I would actually expect that the key `backgroundColor` doesn't exist in `style` since `null` isn't a valid css value for background-color

Reply:
I tried to change this to `expect(Object.keys(wrapper.find('span').props().style).includes('backgroundColor')).to.be.false;` which actually caused the test to fail. I guess because this is shallow rendered the style prop still have `backgroundColor: null` in it?

Reply:
Yeah I think to fix this what you would want to do is only include the `backgroundColor` key in the style object if it's not null in `ProgrammingClassOverview`. That might be overkill though, so feel free to leave it as-is